pnc_cli/productmilestones.py

```diff
@@ -129,6 +129,33 @@ def update_milestone(id, **kwargs):
         return utils.format_json(response.content)
 
 
+@arg("id", help="ProductMilestone ID to update.", type=types.existing_product_milestone)
+@arg("-rd", "--release-date", help="Release date for the ProductMilestone. If not specified, current date is used",
+     type=types.valid_date)
+def close_milestone(id, **kwargs):
+    release_date = kwargs.get('release_date')
+    if not release_date:
+        release_date = datetime.datetime.now()
+
+    existing_milestone = utils.checked_api_call(milestones_api, 'get_specific', id=id).content
+    setattr(existing_milestone, 'end_date', release_date)
+
+    response = utils.checked_api_call(
+        milestones_api, 'update', id=id, body=existing_milestone)
+
+    latest_release = utils.checked_api_call(milestones_api, 'get_latest_release', id=id).content
+
+    while latest_release.status == 'IN_PROGRESS':
+        print("Latest release for milestone is in progress, waiting till it finishes...")
+        time.sleep(60)
+        latest_release = utils.checked_api_call(milestones_api, 'get_latest_release', id=id).content
```

Comment:
I don't like the wait as a default behaviour, as no other REST command does it AFAIK (make-mead does it, but that's not a plain rest command). I think it should be enabled by optional parameter.

Reply:
Required changes done